kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp

```diff
@@ -690,13 +688,11 @@ kvbc::categorization::EventGroup KvbAppFilter::getEventGroup(kvbc::EventGroupId
   return event_group_out;
 }
 
-void KvbAppFilter::setNextExternalEgIdToRead(uint64_t ext_eg_id) { next_ext_eg_id_to_read_ = ext_eg_id; }
-
-uint64_t KvbAppFilter::getNextExternalEgIdToRead() const { return next_ext_eg_id_to_read_; }
-
-void KvbAppFilter::setLastGlobalEgIdReadForClient(uint64_t global_eg_id) { last_global_eg_id_read_ = global_eg_id; }
+void KvbAppFilter::setLastEgIdsRead(std::pair<uint64_t, uint64_t> last_ext_and_global_eg_id_read) {
```

Comment:
Nit/Optional: The fact that the storage format is a pair doesn't need to be part of the function signature. Simply `setLastEgIdsRead(uint64_t external, uint64_t global)` is sufficient.